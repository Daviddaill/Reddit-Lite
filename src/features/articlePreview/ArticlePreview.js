import articleCSS from './articlePreview.module.css'
import Title from '../title/Title';
import Media from '../media/Media';
import Description from '../description/Description';
import SubReddit from '../subReddit/SubReddit';
import { Link } from 'react-router-dom';


function ArticlePreview({id, img, title, description, text, subReddit, comments, vote }) {
  return (
    <div className={articleCSS.card} aria-label='article-preview' data-testid="article-preview">
      <Link to={`/articles/${id-1}`}>
      <Title title={title} className={articleCSS.titleMobile} /> {/* title use only in mobile screen size */}
      <div className={articleCSS.mainDiv}>
        <figure className={articleCSS.section} id={articleCSS.figure}>
          <Media source={img} description={description} className={articleCSS.media} />
        </figure>
        <article className={articleCSS.section} id={articleCSS.article}>
          <Title className={articleCSS.title} title={title} ariaLabel="preview-title" />
          <Description className={articleCSS.description} text={text}  ariaLabel="preview-text"/>
          <SubReddit className={articleCSS.subReddit} vote={vote} comments={comments} subReddit={subReddit} type='preview' />
        </article>
      </div>
      </Link>
      
    </div>
  )
}

export default ArticlePreview;