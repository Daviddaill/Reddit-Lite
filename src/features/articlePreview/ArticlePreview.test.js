import React from 'react';
import { render, screen } from '@testing-library/react';
import ArticlePreview from './ArticlePreview'; // Adjust the import path

describe('ArticlePreview Component', () => {
    const mockProps = {
        img: 'image.jpg',
        title: 'Sample Title',
        description: 'Sample Description',
        text: 'Sample Text',
        subReddit: 'Sample SubReddit',
        comments: '10',
        vote: '5',
    };

    it('should render ArticlePreview component with correct data', () => {
        //Render
        render(<ArticlePreview {...mockProps} />);
        //Find
        const titleElement = screen.queryAllByRole('heading');
        const imgElement= screen.getByRole('img')
        //const textElement= screen.getByLabelText('preview-text');
        const subRedditElement= screen.getByLabelText("subreddit name") 
        const voteElement= screen.getByLabelText("number of votes") 
        const commentsElement= screen.getByLabelText("number of comments") 

        //Assert
        expect(screen.getByTestId('article-preview')).toBeInTheDocument();
        expect(titleElement[1].textContent).toBe(mockProps.title);
        expect(titleElement[2]).not.toBeNull();
        expect(imgElement).toHaveAttribute('src', mockProps.img) 
        //expect(textElement.textContent).toBe(mockProps.text);
        expect(subRedditElement).toHaveTextContent(mockProps.subReddit)
        expect(voteElement).toHaveTextContent(mockProps.vote)
        expect(commentsElement).toHaveTextContent(mockProps.comments) 
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<ArticlePreview {...mockProps} />);
        // Use snapshot testing to ensure component renders consistently
        expect(asFragment()).toMatchSnapshot();
    });
});
