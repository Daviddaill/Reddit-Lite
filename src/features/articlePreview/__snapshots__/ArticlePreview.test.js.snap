// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePreview Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="article-preview"
    class="card"
    data-testid="article-preview"
  >
    <h2
      class="titleMobile"
    >
      Sample Title
    </h2>
     
    <div
      class="mainDiv"
    >
      <figure
        class="section"
        id="figure"
      >
        <img
          alt="Sample Description"
          class="media"
          height="auto"
          src="image.jpg"
          width="98%"
        />
      </figure>
      <article
        class="section"
        id="article"
      >
        <h2
          class="title"
        >
          Sample Title
        </h2>
        <p
          class="description"
        >
          Sample Text
        </p>
        <div
          class="subReddit"
          data-testid="subReddit"
        >
          <div
            aria-label="number of votes"
            class="section"
          >
            <i
              class="fa-solid fa-arrow-up"
            />
            <p>
              5
            </p>
          </div>
          <div
            aria-label="number of comments"
            class="section"
          >
            <i
              class="fa-regular fa-message"
            />
            <p>
              10
            </p>
          </div>
          <div
            aria-label="subreddit name"
            class="section"
          >
            <i
              class="fa-brands fa-reddit-alien"
            />
            <p
              class="name"
            >
              Sample SubReddit
            </p>
          </div>
        </div>
      </article>
    </div>
  </div>
</DocumentFragment>
`;
